Create About Page

UtkarshKr007 opened the pull request:
Closes: #9 and Closes: #182

Commit:
Create about page

Added info icon to sidebar

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>

Commit:
Add enrtf logo image to about page

Co-authored-by: Utkarsh Kumar <kumar375@morris.umn.edu>
Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

UtkarshKr007 committed:
Merge branch 'master' into about-page

UtkarshKr007 committed:
Add a list of contributors to about page

The gatsby-source-github-api doesn't support retrieving contributors
yet. The closest thing it can give is a list of collaborators which are different, so I used a different plugin.
I also filter dependabot out in the GraphQL query. I initally had a
regex query that fitlered out any login with
"bot" in it, but that could theoretically catch future contributors
whose login contain the words 'bot'.

UtkarshKr007 committed:
Add contributor avatar and a fun alternate

I didn't use the alt prop, because in local testing it didn't show to
replace image with first letter of alt vale as material ui docs say it
should.

floogulinc commented on gatsby-config.js:
```diff
@@ -55,6 +55,12 @@ module.exports = {
       },
     },
+    {
+      resolve: `gatsby-source-github-contributors`,
```
We should probably use an API token for GitHub as in Buzz About to prevent throttling issues.

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,101 @@
+            <ListItem button component="a" href={contributor.htmlUrl} key={contributor.login}>
+               <ListItemAvatar>
+                <Avatar src={contributor.avatarUrl} style={{backgroundColor: randomColor()}}>
+                  {displayName.charAt(0).toUpperCase()}
```
Don't do this. Leave the name from GitHub as-is.

floogulinc replied:
The text should be in a `ListItemText` element outside the `ListItemAvatar`. See the [MUI list docs](https://material-ui.com/components/lists/)

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,101 @@
+
+      <List component="nav">
+        {dynamicUsers.nodes.map((contributor) => {
+          const displayName = contributor.name? contributor.name : contributor.login;
```
This can just be
```suggestion
          const displayName = contributor.name ?? contributor.login;
```

UtkarshKr007 replied:
Oh, cool. I didn't know there was a shorthand for that. It wouldn't let me commit suggestion due to 'outdatedness' but, I will make that change along with the background color in next commit.

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,101 @@
+          return (
+            <ListItem button component="a" href={contributor.htmlUrl} key={contributor.login}>
+               <ListItemAvatar>
+                <Avatar src={contributor.avatarUrl} style={{backgroundColor: randomColor()}}>
```
What is this `randomColor` thing for?

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,101 @@
+          return (
+            <ListItem button component="a" href={contributor.htmlUrl} key={contributor.login}>
+               <ListItemAvatar>
+                <Avatar src={contributor.avatarUrl} style={{backgroundColor: randomColor()}}>
```
Instead of doing this random color and using the first letter of the username, we can probably just assume there's always an avatar from GitHub. Even users without a set avatar are assigned a generated one. Setting the background color here seems to cause some styling issues when the actual images are placed in, there is sometimes a slight border around them of this random background color.

For fallbacks when there isn't an avatar somehow, we should use the built-in fallback options provided by MUI: https://material-ui.com/components/avatars/#fallbacks

I think it will automatically use the first letter of the alt text or just use the generic fallback avatar.

UtkarshKr007 replied:
I was testing those fallbacks methods, but for some reason. I never saw it use first letter of alt text and instead always fall to the generic avatar icon, but yeah I will remove the random color + text combo. I forgot that github assigns a generated avatar as well and having random colors around the borders is definitely undesirable..

UtkarshKr007 committed:
Put Contributors in Card and style fundings better

UtkarshKr007 committed:
Merge branch 'master' into about-page

UtkarshKr007 committed:
Remove color + letter avatar alternate

UtkarshKr007 committed:
Add card about MN Pollinator and link github org

UtkarshKr007 committed:
Add the License card with temporary wording.

UtkarshKr007 committed:
Add in information about Pollinator Facts

floogulinc commented on src/pages/about.js:
```diff
@@ -15,9 +17,36 @@ import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+    gridTemplateColumns: '100%'
+  },
+  aboutCards: {
+    maxWidth: '800px',
```
Shouldn't we just let the cards here fill the regular max width we have for everything on this site?

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,191 @@
+  enrtfLogo: file(relativePath: { eq: "enrtf_logo.png" }) {
+    childImageSharp {
+			fixed(width: 180) {
+				...GatsbyImageSharpFixed_withWebp
```
The blur effect really doesn't work well on the logo so I think we should do the same as on the footer and disable it.

```suggestion
				...GatsbyImageSharpFixed_withWebp_noBase64
```

UtkarshKr007 committed:
Update src/pages/about.js

Remove blur on logo

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

UtkarshKr007 committed:
Remove maxWidth from cards

UtkarshKr007 committed:
Add in more contributors

UtkarshKr007 committed:
Change token variable name

UtkarshKr007 committed:
Revert "Change token var to match netlify"

This reverts commit 85815f59720f781749fea946590ff86661a71ffa.

UtkarshKr007 committed:
Change token var to match netlify

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+
+          <Card>
+            <CardContent>
+              <Typography gutterBottom variant="h5" component="h2">Pollinator Facts</Typography>
```
Can we use the `CardHeader` component to do the same type of header with the icon as in Buzz About?

eg:
![image](https://user-images.githubusercontent.com/1300395/102800555-a7305b00-4379-11eb-98bf-327ef376ae79.png)

UtkarshKr007 replied:
I just added the green plant icon we already had in pollinator-fact repository to the about page. I can switch it to the icon on buzz-about if we want to be consistent. In that case, it would be nice to have a green version of that bee icon to match the theme of pollinator-facts.

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+              <Typography>
+                Pollinator Facts website was created alongside the <a href="https://buzzabout.app">Buzz About</a> participatory simulation game to
+                teach people about native pollinators of Minnesota. This website holds many interesting tidbits as well beautiful pictures of pollinators, their food, homes
+                and habits.
```
We probably want to work on this blurb. I don't know if we want to link to Buzz About here.

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+			}
+    }
+  }
+  dynamicUsers: allGitHubContributor(sort: {fields: contributions, order: DESC}, filter: {login: {ne: "dependabot-preview[bot]"}}) {
```
Can we remove the `order:desc` here so it just uses the default ordering by contributions?

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+                  return (
+                    <ListItem button component="a" href={contributor.htmlUrl} key={contributor.login}>
+                      <ListItemAvatar>
+                        <Avatar alt={displayName} src={contributor.avatarUrl} />
```
We probably want to use a `srcSet` with a few different sizes of the avatar. 

How it's being done in Buzz About:
```html
<img matListAvatar src="{{contributor.avatarUrl}}&size=40" srcset="{{contributor.avatarUrl}}&size=80 2x, {{contributor.avatarUrl}}&size=160 4x">
```

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+                  )
+                })}
+
+                {Object.keys(staticUserList).map((objectName, index) => {
```
Why not use `Object.values` here?

UtkarshKr007 replied:
[Object.values()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values#Browser_compatibility) is not supported on Internet Explorer while [.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#Browser_compatibility) is.

floogulinc replied:
That shouldn't be an issue since Gatsby uses Babel and will polyfill newer features when needed.

floogulinc commented on gatsby-config.js:
```diff
@@ -55,6 +59,41 @@ module.exports = {
+    {
+      resolve: `gatsby-source-github-contributors`,
+      options: {
+        repo: "mn-pollinators/pollinator-facts"
```
```suggestion
        repo: "mn-pollinators/pollinator-facts",
        token: process.env.GITHUB_TOKEN
```

floogulinc commented on src/pages/about.js:
```diff
@@ -0,0 +1,229 @@
+            <CardContent>
+              <Typography gutterBottom variant="h5" component="h2">License</Typography>
+              <Typography>
+                The images used for facts belong to their respective owners. Clicking the info button on the bottom right of a fact
+                presents the information and links to the image owners as well as the license under which the image is published.
```
Definitely need to work on this.

floogulinc reviewed (changes requested):
Looking pretty good, just a few concerns.

UtkarshKr007 committed:
Add icon on about page and give token to plugin

GatsbyImage doesn't support SVG format, so we have to import the svg
into the about.js file and then use it.

UtkarshKr007 committed:
Use srcSet for GH avatars.Remove buzz-about link

UtkarshKr007 committed:
Use Object.values()

UtkarshKr007 committed:
Remove manual sort on GraphQL query

UtkarshKr007 committed:
Move repeated contributor list code to component
